consensus/src/lib.rs

```diff
@@ -173,8 +179,9 @@ pub fn finalize_consensus_data(
     chain_config: &ChainConfig,
     block_header: &mut BlockHeader,
     block_height: BlockHeight,
+    block_epoch: &Arc<AtomicU64>,
```

Comment:
Passing a reference of an Arc is bad practice. If you can simply call as_ref on the Arc and pass &AtomicU64. Same lifetime.

Reply:
Thanks for pointing that out. Fixed.